### DISCUSSION TITLE
Low memory, PHP7.4 setups

### DISCUSSION THREAD

**Discussion by sunshinephotocart** [maennchen/ZipStream-PHP#287] [Q&A] [answered]
I am trying to use this in a WordPress plugin where there are low end servers using PHP 7.4. I am using the ZipStream 2.2.6 to work with PHP 7.4.

Is there a way to use this class to use less memory? I have instances where we are adding say 50 x 4MB images and it will run out of PHP memory and the zip file fails.

**Comment by NicolasCARPi** [comment]
Are you streaming the files from storage to the library (and possibly to the http response)?

**Comment by wpsunshine** [comment]
I have two separate scenarios where I am building the zip file from files on the server or being pulled from S3. Right now I’m just testing local files as my current implementation pulls them from S3 to local first and then builds the zip, but ideally would like a solution that pulls directly from S3 eventually.

**Reply by NicolasCARPi** [reply]
You can stream from local or s3 to the browser while zipping on the fly, that's what my app do, with the help of flysystem, symfony, and zipstream-php.

**Reply by sunshinephotocart** [reply]
I am testing using ZipStream-PHP already and have it working - but I am running into memory issues when building from local files. I was wondering if there was a setting or something I am missing that let's ZipStream-PHP run better in a low memory environment, even if it runs slower. Since I am making a WordPress plugin, I do not have control over the server config and have to work in low memory and resource offerings.

**Reply by NicolasCARPi** [reply, marked as answer]
Look at addFileFromStream().
